spec/models/user_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :user do
+  before(:context) do
+    @user = User.create(name: 'George R Martin', photo: '', bio: 'https//photo.png', posts_counter: 0)
+  end
+
+  describe 'Validations' do
+    it 'name must not be blank' do
+      @user.name = nil
+      expect(@user).to_not be_valid
+    end
+
+    it 'posts_counter must not be a non-integer' do
+      @user.posts_counter = 'five'
+      expect(@user).to_not be_valid
+    end
+
+    it 'posts_counter must be an integer greater or equal to 0' do
+      @user.posts_counter = -1
+      expect(@user).to_not be_valid
+    end
+  end
+
+  describe 'scopes' do
+    it 'recent posts count should be 3' do
+      Post.create(author: @user, title: 'post 1', text: 'content', comments_counter: 3, likes_counter: 3)
+      Post.create(author: @user, title: 'post 2', text: 'content', comments_counter: 3, likes_counter: 3)
+      Post.create(author: @user, title: 'post 3', text: 'content', comments_counter: 3, likes_counter: 3)
+      Post.create(author: @user, title: 'post 3', text: 'content', comments_counter: 3, likes_counter: 3)
+      expect(User.recent_posts(@user).count).to eq(3)
+    end
```

Comment:
- [ ] According to the project requirements, Any custom methods in the models should have a corresponding unit test. Kindly make sure the `update_posts_counter` and `recent_comments` methods from the Post model have a corresponding unit test. ✍🏻

Reply:
Thanks for the feedback @BenMukebo.